hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -210,15 +214,33 @@ private HoodieLogBlock readBlock() throws IOException {
   private HoodieLogBlock createCorruptBlock() throws IOException {
     log.info("Log " + logFile + " has a corrupted block at " + inputStream.getPos());
     long currentPos = inputStream.getPos();
-    long nextBlockOffset = scanForNextAvailableBlockOffset();
+    byte[] corruptedBytes;
+    int corruptedBlockSize = 0;
+    boolean done = true;
+    BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+    do {
+      // read upto logblocksize to find next magic header
+      corruptedBytes = new byte[logBlockSize];
+      int numberOfBytesRead = bufferedInputStream.read(corruptedBytes, 0, logBlockSize);
+      int estimatedCorruptBlockSize = (int) scanForNextAvailableBlockOffset(corruptedBytes,
+          numberOfBytesRead);
+      if (numberOfBytesRead == logBlockSize && estimatedCorruptBlockSize == numberOfBytesRead) {
+        // there can be more data to read, possibly back to back corrupt blocks
+        // or different sized log block than logBlockSize
+        done = false;
+      }
+      corruptedBlockSize += estimatedCorruptBlockSize;
+    } while (!done);
```

Comment:
so we return 1 corrupt block if there are back-back corrupt blocks?

Reply:
Yes, ideally there shouldn't be back to back corrupt blocks but if it failed to write the magic header for the next block too..